src/lambda_function/events/generate_test_event.py

```diff
@@ -0,0 +1,138 @@
+    parser.add_argument("--input-bucket",
+      default="recover-dev-input-data",
+      help="Optional. S3 bucket name containing input data. Default is 'recover-dev-input-data'.")
+    parser.add_argument("--input-key",
+      default="main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368",
+      help=("Optional. Name of the Synapse dataset containing test data. "
+            "Default is `main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368` "))
+    parser.add_argument("--input-key-prefix",
+      help=("Optional. Takes precedence over `--input-key`. If you want to generate a "
+            "single event containing all data under a specific S3 key prefix, "
+            "specify that here."))
+    args = parser.parse_args()
+    return args
+
+def create_event(bucket: str, key: str, key_prefix: str) -> dict:
+    """
+    Create an S3 event notification for testing.
+
+    This function accepts either an S3 object key or an S3 key prefix that will
+    be included in the test event. If an S3 object key is provided, then the test
+    event will include only that S3 object. If an S3 key prefix is provided, the
+    test event will include every S3 object that is directly under that prefix.
+    The key prefix behavior takes precedence over the single key behavior.
+
+    Args:
+        bucket (str): The S3 bucket name
+        key (str): An S3 object key
+        key_prefix (str): An S3 key prefix containing S3 objects to include
+            in the test event. Takes precedence over `key`.
+
+    Returns:
+        dict: An S3 event notification
+    """
+    if key_prefix is not None:
+        s3_client = boto3.client("s3")
+        test_objects = s3_client.list_objects_v2(
+                Bucket=bucket,
+                Prefix=key_prefix
+        )
+        test_data = [obj["Key"] for obj in test_objects["Contents"]]
+    else:
+        test_data = [key]
+    s3_event = {
+            "Records": [create_s3_event_record(bucket=bucket, key=k) for k in test_data]
+    }
+    return s3_event
+
+def create_s3_event_record(bucket: str, key: str) -> dict:
+    """
+    Create an S3 event notification "Record" for an individual S3 object.
+
+    Args:
+        bucket (str): The S3 bucket name
+        key (str): The S3 object key
+
+    Returns:
+        dict: A dictionary formatted as a "Record" object would be
+            in an S3 event notification
+    """
+    s3_record = copy.deepcopy(s3_event_record_template)
+    s3_record["s3"]["bucket"]["name"] = bucket
+    s3_record["s3"]["object"]["key"] = key
+    return s3_record
+
+def main() -> None:
+    args = read_args()
+    print("Generating mock S3 event...")
+    s3_event = create_event(
+            bucket=args.input_bucket,
+            key=args.input_key,
+            key_prefix=args.input_key_prefix
+    )
+
+    if args.input_key_prefix is not None:
+        with open(MULTI_RECORD_OUTFILE, "w") as outfile:
```

Comment:
Since we don't expect this script to live on forever, its probably fine to not have these as parameters.